From b29712e941e15c8385cf8f0347b165c148b133bc Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Fri, 27 Jul 2018 23:50:53 +0800
Subject: [PATCH 15/32] video: sunxi: de2: fix SimpleFB node creation when DE2
 not probed

Sometimes when a monitor without EDID information is plugged, the DE2
won't be probed (because of lack of timing information), but the HDMI
node is probed, thus a SimpleFB node with invalid information will be
populated.

Also detect whether DE2 is probed when creating SimpleFB node.

Fixes: be5b96f0e411 ("sunxi: setup simplefb for Allwinner DE2")
Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
Acked-by: Maxime Ripard <maxime.ripard@bootlin.com>
Acked-by: Jagan Teki <jagan@openedev.com>
---
 drivers/video/sunxi/sunxi_de2.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/video/sunxi/sunxi_de2.c b/drivers/video/sunxi/sunxi_de2.c
index 4ed035d556..8333ddc44c 100644
--- a/drivers/video/sunxi/sunxi_de2.c
+++ b/drivers/video/sunxi/sunxi_de2.c
@@ -347,6 +347,9 @@ int sunxi_simplefb_setup(void *blob)
 	if (ret) {
 		debug("DE2 not present\n");
 		return 0;
+	} else if (!device_active(de2)) {
+		debug("DE2 present but not probed\n");
+		return 0;
 	}
 
 	ret = uclass_find_device_by_name(UCLASS_DISPLAY,
-- 
2.18.0

